engine/src/main/java/org/pentaho/di/trans/steps/file/BaseFileInputAdditionalField.java

```diff
@@ -0,0 +1,90 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2017 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.file;
+
+import org.apache.commons.lang.StringUtils;
+import org.pentaho.di.core.injection.Injection;
+
+/**
+ * Additional fields settings.
+ */
+public class BaseFileInputAdditionalField implements Cloneable {
+
+  /** Additional fields **/
+  @Injection( name = "FILE_SHORT_FILE_FIELDNAME" )
+  public String shortFilenameField;
+  @Injection( name = "FILE_EXTENSION_FIELDNAME" )
+  public String extensionField;
+  @Injection( name = "FILE_PATH_FIELDNAME" )
+  public String pathField;
+  @Injection( name = "FILE_SIZE_FIELDNAME" )
+  public String sizeField;
+  @Injection( name = "FILE_HIDDEN_FIELDNAME" )
+  public String hiddenField;
+  @Injection( name = "FILE_LAST_MODIFICATION_FIELDNAME" )
+  public String lastModificationField;
+  @Injection( name = "FILE_URI_FIELDNAME" )
+  public String uriField;
+  @Injection( name = "FILE_ROOT_URI_FIELDNAME" )
+  public String rootUriField;
+
+  @Override
+  public Object clone() {
+    try {
+      return super.clone();
+    } catch ( CloneNotSupportedException ex ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)